specifyweb/specify/migrations/0002_geo.py

```diff
@@ -0,0 +1,276 @@
+                ('cogtype', models.ForeignKey(db_column='COGTypeID', on_delete=protect_with_blockers, related_name='collectionobjectgroups', to='specify.collectionobjectgrouptype')),
+                ('createdbyagent', models.ForeignKey(db_column='CreatedByAgentID', null=True, on_delete=protect_with_blockers, related_name='+', to='specify.agent')),
+                ('modifiedbyagent', models.ForeignKey(db_column='ModifiedByAgentID', null=True, on_delete=protect_with_blockers, related_name='+', to='specify.agent')),
+            ],
+            options={
+                'db_table': 'collectionobjectgroup',
+                'ordering': (),
+            },
+        ),
+        migrations.CreateModel(
+            name='CollectionObjectGroupJoin',
+            fields=[
+                ('id', models.AutoField(db_column='collectionobjectgroupjoinid', primary_key=True, serialize=False)),
+                ('isprimary', models.BooleanField(blank=True, db_column='IsPrimary', null=True)),
+                ('issubstrate', models.BooleanField(blank=True, db_column='IsSubstrate', null=True)),
+                ('precedence', models.SmallIntegerField(blank=True, db_column='Precedence', null=True)),
+                ('version', models.IntegerField(blank=True, db_column='Version', default=0, null=True)),
+                ('timestampcreated', models.DateTimeField(db_column='TimestampCreated', default=django.utils.timezone.now)),
+                ('timestampmodified', models.DateTimeField(blank=True, db_column='TimestampModified', default=django.utils.timezone.now, null=True)),
+                ('text1', models.TextField(blank=True, db_column='Text1', null=True)),
+                ('text2', models.TextField(blank=True, db_column='Text2', null=True)),
+                ('text3', models.TextField(blank=True, db_column='Text3', null=True)),
+                ('integer1', models.IntegerField(blank=True, db_column='Integer1', null=True)),
+                ('integer2', models.IntegerField(blank=True, db_column='Integer2', null=True)),
+                ('integer3', models.IntegerField(blank=True, db_column='Integer3', null=True)),
+                ('yesno1', models.BooleanField(blank=True, db_column='YesNo1', null=True)),
+                ('yesno2', models.BooleanField(blank=True, db_column='YesNo2', null=True)),
+                ('yesno3', models.BooleanField(blank=True, db_column='YesNo3', null=True)),
+                ('parentcog', models.ForeignKey(db_column='ParentCOGID', on_delete=django.db.models.deletion.CASCADE, related_name='parentcojos', to='specify.collectionobjectgroup')),
+                ('childcog', models.OneToOneField(db_column='ChildCOGID', null=True, on_delete=django.db.models.deletion.CASCADE, related_name='cojo', to='specify.collectionobjectgroup')),
+                ('childco', models.OneToOneField(db_column='ChildCOID', null=True, on_delete=django.db.models.deletion.CASCADE, related_name='cojo', to='specify.collectionobject')),
```

Comment:
This should also not be a one-to-one.... you won't be able to add another collection object to the group?

On a side-note: Wow, reviewing the migration file is much easier than actual models

Reply:
basically reverse of the above. when you add a new child cog, it'll error out. Or, are we planning to have them set to null for both of these to avoid that constraint? If yes, then that's awkward, but fine